### PR TITLE
Add some information about the dev process.

### PR DESCRIPTION
**Pull request by Carreau**

### CODE REVIEW

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 1]
```diff
@@ -39,6 +40,12 @@ below to get started. If you experience any problems while working on the issue,
 leave a comment on the issue page in GitHub and someone on the core team will
 be able to lend you assistance.
 
+Keep in mind that all you see below is just a guide, that describe how things
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: Please keep in mind that what follows are guidelines. If you work through the steps and have questions or run into time constraints, please submit what you already have worked on as a pull request and ask questions on it. Your effort, including partial or in-progress work, is appreciated.

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 2]
```diff
@@ -55,20 +62,26 @@ be able to lend you assistance.
...
-5. Open a pull request to the repository with [WIP] appended to the front
+5. Open a pull request to the repository with ``[WIP]`` appended to the front
    so that the core team is aware that you are actively pursuing the issue.
    When creating a pull request, make sure that the title clearly and concisely
    described what your code does. For example, we might use the title "Updated
    error message on ExampleException". In the body of the pull request, make sure
    that you include the phrase "Closes #issue-number-here", where the issue number is
    the issue number of the issue that you are addressing in this PR.
 
+   Feel free to open the PR as early as possible, you can get sometime early
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: Feel free to open a PR as early as possible. Getting early feedback on your approach will save you time and prevent the need for an extensive refactor later.

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 3]
```diff
@@ -55,20 +62,26 @@ be able to lend you assistance.
...
+   feedback on your approach, which might be more efficient that polishing a
+   Pull requests that would need extensive refactor afterward.
+
 6. Run the test suite locally in order to ensure that everything is properly
    configured on your system. Refer to the repository's README for information
    on how to run the test suite. This will typically require that you run the
-   ``nosetests`` command on the commandline.
+   ``nosetests`` command on the commandline. Alternatively, you can just submit
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: Alternatively, you may submit a pull request. Our Continuous Integration system will test your code and report test results.

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 4]
```diff
@@ -109,3 +122,37 @@ be able to lend you assistance.
 Contribution Workflow
 ---------------------
 .. image:: ../_static/_images/contribution_workflow.png
+
+
+
+Core Developer Workflow
+-----------------------
+
+It is interesting to know how the core developer works in order to understand
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: To help you understand our review process by core developers after you submit a pull request, here's a guide that outlines the general process (specifics may vary a bit across our repositories). Here is an example for Jupyter notebook 4.x:

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 5]
```diff
@@ -109,3 +122,37 @@ be able to lend you assistance.
...
+some of the reason for some requests. This is again a rough guide, the exact
+process might change a bit across our repositories. Here is an example [for Jupyter/notebook 4.x].
+
+
+In general, Pull-Retests are against ``master`` unless they only affect a backport
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: Pull Requests

File: docs/source/developer-docs/contrib_guide_code.rst [hunk 6]
```diff
@@ -109,3 +122,37 @@ be able to lend you assistance.
...
+branch. If a PR affects master and should be backported, the general flow is:
+
+  0. mark the PR with milestone for the next backport release (4.3)
+  1. merge into master
+  2. backport to 4.x
+  3. push updated 4.x branch
+
+Backports can be done in a variety of ways, but we have `a script
+<https://github.com/ipython/ipython/blob/master/tools/backport_pr.py>`_ for
+automating the common process of:
```
Comment (willingc) [inline docs/source/developer-docs/contrib_guide_code.rst]: of: -> to:

**Comment by willingc** [comment]
@carreau Looks good. Here are a few edits. +1 to merging after edits :smile:

P.S. Japan is beautiful.

**Comment by damianavila** [comment]
>+1 to merging after edits :smile:

:+1: 

>P.S. Japan is beautiful.

I have no doubts about it :wink:

**Comment by Carreau** [comment]
Hope you are still enjoying Japan. 

Comments adressed,

**Pull request merged**